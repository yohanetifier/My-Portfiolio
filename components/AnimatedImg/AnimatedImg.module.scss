// .imgWrapper {
//     border: 2px solid purple;
//     position: absolute;
//     width: 200px;
//     height: 300px;
//     transform: rotate(50deg);
// }

.container {
    border: 5px solid black;
    position: absolute;
    width: 100vw;
    height: 100vh;
    display: flex;
    justify-content: center;
    align-items: center;
}
