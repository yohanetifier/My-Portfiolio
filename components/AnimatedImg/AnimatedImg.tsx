import { Variants, motion } from 'framer-motion';
import Image from 'next/image';
import React from 'react';
import styles from './AnimatedImg.module.scss';
import { container } from '../AnimatedWords/AnimatedWords';

export type ArrayOfImg = {
	src: string;
	alt: string;
	className: string;
};

type Props = {
	arrayOfImg: ArrayOfImg[];
	animationComplete: boolean;
	setAnimationSecondComplete: (arg: boolean) => void;
};

const imageVariants: Variants = {
	hidden: { opacity: 0, transition: { duration: 0.3 } },
	show: { opacity: 1, transition: { duration: 0.3 } },
};

const AnimatedImg = ({
	arrayOfImg,
	animationComplete,
	setAnimationSecondComplete,
}: Props) => {
	return (
		<motion.div
			className={styles.container}
			variants={container}
			initial='hidden'
			animate={animationComplete ? 'show' : ''}
			onAnimationComplete={() => setAnimationSecondComplete(true)}
		>
			{arrayOfImg.map(({ src, alt, className }, i) => (
				<>
					<motion.figure
						className={`${className} figure`}
						key={i}
						variants={imageVariants}
					>
						<Image
							src={src}
							alt={alt}
						/>
					</motion.figure>
				</>
			))}
		</motion.div>
	);
};

export default AnimatedImg;
