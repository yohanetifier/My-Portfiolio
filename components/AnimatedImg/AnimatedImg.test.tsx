import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import AnimatedImg from './AnimatedImg';

describe('AnimatedImg', () => {
	it('should render correctly', () => {
		const arrayOfImg = [
			{
				id: 1,
				src: '/exemple.png',
				alt: 'exemple',
				className: 'exemple',
			},
			{
				id: 2,
				src: '/test.png',
				alt: 'test',
				className: 'test',
			},
		];
		const { container } = render(
			<AnimatedImg
				arrayOfImg={arrayOfImg}
				animationComplete={false}
				setHideIntro={() => {}}
			/>,
		);
		expect(container).toMatchSnapshot();
		const figureList = screen.getAllByRole('figure');
		const imageList = screen.getAllByRole('img');
		expect(figureList).toHaveLength(2);
		expect(imageList).toHaveLength(2);
		/* Check class */
		expect(figureList[0]).toHaveClass('exemple');
		expect(figureList[1]).toHaveClass('test');
		/* Check props */
		expect(imageList[1]).toHaveProperty('src');
		expect(imageList[1]).toHaveProperty('alt');
		expect(imageList[1]).toHaveProperty('className');
	});
});
