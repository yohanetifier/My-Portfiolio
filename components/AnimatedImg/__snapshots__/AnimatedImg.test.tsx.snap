// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedImg should render correctly 1`] = `
<div>
  <div
    class="container"
  >
    <figure
      class="exemple figure"
    >
      <img
        alt="exemple"
        data-nimg="1"
        decoding="async"
        height="200"
        loading="lazy"
        src="/_next/image?url=%2Fexemple.png&w=640&q=75"
        srcset="/_next/image?url=%2Fexemple.png&w=256&q=75 1x, /_next/image?url=%2Fexemple.png&w=640&q=75 2x"
        style="color: transparent;"
        width="200"
      />
    </figure>
    <figure
      class="test figure"
    >
      <img
        alt="test"
        data-nimg="1"
        decoding="async"
        height="200"
        loading="lazy"
        src="/_next/image?url=%2Ftest.png&w=640&q=75"
        srcset="/_next/image?url=%2Ftest.png&w=256&q=75 1x, /_next/image?url=%2Ftest.png&w=640&q=75 2x"
        style="color: transparent;"
        width="200"
      />
    </figure>
  </div>
</div>
`;
